src/db/pool.js

```
@@ -0,0 +1,95 @@
+import { logger } from '../config/logger.js';
+
+const { Pool } = pg;
+
+/**
+ * PostgreSQL connection pool
+ * Singleton pattern to ensure single pool instance across the application
+ */
+class DatabasePool {
+  constructor() {
+    if (DatabasePool.instance) {
+      return DatabasePool.instance;
```

Biome (2.1.2)

[error] The constructor should not return a value.

The constructor is here:

Returning a value from a constructor may confuse users of the class.

(lint/correctness/noConstructorReturn)